lnbits/wallets/breez.py

```diff
@@ -0,0 +1,257 @@
+                    breez_sdk.ReceivePaymentRequest(
+                        amount * 1000,  # breez uses msat
+                        unhashed_description.decode() if unhashed_description else memo,
+                        preimage=kwargs.get("preimage"),
+                        opening_fee_params=None,
+                        use_description_hash=True if unhashed_description else None,
+                    )
+                )
+            )
+
+            return InvoiceResponse(
+                True,
+                breez_invoice.ln_invoice.payment_hash,
+                breez_invoice.ln_invoice.bolt11,
+                None,
+            )
+
+        async def pay_invoice(
+            self, bolt11: str, fee_limit_msat: int
+        ) -> PaymentResponse:
+            invoice = lnbits_bolt11.decode(bolt11)
+
+            try:
+                send_payment_request = breez_sdk.SendPaymentRequest(bolt11=bolt11)
+                send_payment_response: breez_sdk.SendPaymentResponse = (
+                    self.sdk_services.send_payment(send_payment_request)
+                )
+                payment: breez_sdk.Payment = send_payment_response.payment
+            except Exception as exc:
+                logger.info(exc)
+                try:
+                    # try to report issue to Breez to improve LSP routing
+                    self.sdk_services.report_issue(
+                        breez_sdk.ReportIssueRequest.PAYMENT_FAILURE(
+                            breez_sdk.ReportPaymentFailureDetails(invoice.payment_hash)
+                        )
+                    )
+                except Exception as ex:
+                    logger.info(ex)
+                # assume that payment failed?
+                return PaymentResponse(
+                    False, None, None, None, f"payment failed: {exc}"
+                )
+
+            assert (
+                payment.status == breez_sdk.PaymentStatus.COMPLETE
+            ), "payment is pending"
+            # let's use the payment_hash as the checking_id
+            checking_id = invoice.payment_hash
+
+            return PaymentResponse(
+                True,
+                checking_id,
+                payment.fee_msat,
+                payment.details.data.payment_preimage,
+                None,
+            )
+
+        async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+            payment: breez_sdk.Payment = self.sdk_services.payment_by_hash(checking_id)
+            if payment is None:
+                return PaymentPendingStatus()
+            if payment.payment_type != breez_sdk.PaymentType.RECEIVED:
+                raise ValueError(f"unexpected payment type: {payment.status}")
+            if payment.status == breez_sdk.PaymentStatus.COMPLETE:
+                return PaymentSuccessStatus()
+            return PaymentPendingStatus()
+
+        async def get_payment_status(self, checking_id: str) -> PaymentStatus:
```

Comment:
- add `try-except` (for rpc errors, etc) and return `PaymentPendingStatus()` on `except`
- do not raise, but instead log and `return PaymentPendingStatus()`